util/Pos/Util/DynamicTimer.hs

```diff
@@ -0,0 +1,36 @@
+-- | Restartable, STM-based dynamic timer build on top of `Pos.Util.Timer.Timer`.
+module Pos.Util.DynamicTimer
+  ( DynamicTimer
+  , newDynamicTimer
+  , waitDynamicTimer
+  , startDynamicTimer
+  ) where
+
+import           Data.Time.Units (Microsecond, toMicroseconds)
+import           Control.Concurrent.STM (readTVar, registerDelay, retry)
+import           Universum
```

Comment:
Let's use the `STM` definitions, not the `Universum` ones (`newTVarIO`). We ought to divest from `Universum`.